File: app/backend/src/frameworks_and_drivers/persistence/entities/color_entity.rs
use sqlx::FromRow;

/// カラーエンティティ
/// Clean Architecture: Frameworks & Drivers層のデータ表現
#[derive(Debug, FromRow)]
pub struct ColorEntity {
    pub id: i64,                           // INTEGER PK
    pub name: String,
    pub hex: String,
    pub created_at: String,
    pub updated_at: String,
} 

File: app/backend-ts/src/infrastructure/database/entities/color.entity.ts
import {
  Entity,
  Column,
  PrimaryColumn,
  CreateDateColumn,
  UpdateDateColumn,
} from 'typeorm';

@Entity('colors')
export class ColorEntity {
  @PrimaryColumn('integer')
  id: number;

  @Column('text')
  name: string;

  @Column('text')
  hex: string;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;
}
